agent/command/expansion_update.go

```diff
@@ -73,6 +73,7 @@ func (c *update) ExecuteUpdates(ctx context.Context, conf *internal.TaskConfig)
 				return errors.WithStack(err)
 			}
 			conf.Expansions.Put(update.Key, newValue)
+			conf.DynamicExpansions.Put(update.Key, newValue)
```

Comment:
Since `expansions.update` is the only usage of `DynamicExpansions`, can we have the command initialize `DynamicExpansions` in the case that it's nil? This also protects the agent from potential panics, where it assumes the `DynamicExpansions` exist.

Alternatively, explicitly testing a code path that runs `NewTaskConfig` and asserting that it must return a non-nil `DynamicExpansions` after initialization may be a reasonable substitute.

Reply:
added to setup test

Reply:
I still have this question, I meant adding something like this to the command's `Execute`, not just changing the test setup:
```
func Execute(...) {
// ...
if conf.DynamicExpansions == nil {
    // Initialize the map
}
// Actually do the expansions.update logic
}
```